plasmapy/utils/pytest_helpers.py

```diff
@@ -719,3 +720,145 @@ def run_test_equivalent_calls(*test_inputs, require_same_type: bool = True):
+        elif param_name == "coulomb_log":
+            unit = 1.0
+            magnitude = 1e1
+        elif param_name == "characteristic_length":
+            unit = u.m
+            magnitude = 1.0
+
+        # else, last resort, if it has a default argument, go with that:
+        elif not (param_default is inspect._empty):
+            return param_default, param_default, param_default
+
+        else:
+            raise ValueError("Unrecognized function input")
+
+        # now knowing unit and magnitude, set up the 0d, 1d, and 2d arrays:
+        input_data_2d = np.reshape(np.arange(1.0, 5.0, 1.0), (2, 2))
+        input_data_1d = np.arange(1.0, 5.0, 1.0)
+        if param_name in insert_some_nans:
+            input_data_2d[0, 1] = np.nan
+            input_data_2d[1, 0] = np.nan
+            input_data_1d[1] = np.nan
+        elif param_name in insert_all_nans:
+            input_data_2d = np.ones((2, 2)) * np.nan
+            input_data_1d = np.ones(4) * np.nan
+        input_data_2d *= magnitude
+        input_data_2d *= unit
+        input_data_1d *= magnitude
+        input_data_1d *= unit
+        input_data_0d = input_data_1d[3]
+        return input_data_0d, input_data_1d, input_data_2d
+
+    #
+    # *** body of assert_can_handle_nparray function ***
+    #
+
+    # call _prepare_input to prepare 0d, 1d, and 2d sets of arguments for the function:
+    function_sig = inspect.signature(function_to_test)
+    function_params = function_sig.parameters
+    args_0d = dict()
+    args_1d = dict()
+    args_2d = dict()
+    param_names = [elm for elm in function_params.keys()]
+    for idx, key in enumerate(function_params):
+        args_0d[key], args_1d[key], args_2d[key] = _prepare_input(param_names[idx],
+                                                                  function_params[key].default,
+                                                                  insert_some_nans,
+                                                                  insert_all_nans,
+                                                                  kwargs)
+
+    # call the function with the prepared argument sets:
+    result_0d = function_to_test(**args_0d)
+    result_1d = function_to_test(**args_1d)
+    result_2d = function_to_test(**args_2d)
+
+    # assert that the 1d, 2d versions get the same result (elementwise) as the 0d version:
+    # (if the function returns multiple values, loop through and test each)
+    try:
+        scalar_testable = result_0d.value
+    except AttributeError:
+        scalar_testable = result_0d
+    if np.isscalar(scalar_testable):
+        astrohelper.assert_quantity_allclose(result_0d, result_1d[3])
+        astrohelper.assert_quantity_allclose(result_0d, result_2d[1, 1])
+    else:
+        for idx, res_0d in enumerate(result_0d):
+            astrohelper.assert_quantity_allclose(res_0d, result_1d[idx][3])
+            astrohelper.assert_quantity_allclose(res_0d, result_2d[idx][1, 1])
```

Comment:
I would love to have an idea for how to make this have shared setup, be as easy to call as you've made it (:clap: ) and still get us independent result for cleaner pytest results. I don't, right now.

Reply:
Thanks!  Yeah... I was hoping to find a nice out-of-the-box pytest way to have it do what I wanted, but, I couldn't quite figure it out, so I have this thing at the moment. I'm 100% open to suggestions.